File: app/src/main/java/gardezi/io/amiibofinder/data/AmiiboApi.java
package gardezi.io.amiibofinder.data;


import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface AmiiboApi {

    String HTTPS_API_AMIIBO_URL = "http://www.amiiboapi.com/api/";

    @GET("amiibo/")
    Single<AmiiboPayload> getAmiiboByName(@Query("name") String name);

}

File: app/src/main/java/gardezi/io/amiibofinder/data/AmiiboApi.kt
package gardezi.io.amiibofinder.data

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface AmiiboApi {

    @GET("amiibo/")
    fun getAmiiboByName(@Query("name") name: String): Single<AmiiboPayload>

    companion object {
        val baseUrl = "http://www.amiiboapi.com/api/"
    }

}
